src/commonMain/kotlin/fr/acinq/lightning/channel/ChannelData.kt

```diff
@@ -417,8 +420,16 @@ object ChannelFlags {
 
 data class ClosingTxProposed(val unsignedTx: ClosingTx, val localClosingSigned: ClosingSigned)
 
-/** This gives the reason for creating a new channel. */
-sealed class ChannelOrigin {
-    data class PayToOpenOrigin(val paymentHash: ByteVector32, val fee: Satoshi) : ChannelOrigin()
-    data class PleaseOpenChannelOrigin(val requestId: ByteVector32, val serviceFee: MilliSatoshi, val fundingFee: Satoshi) : ChannelOrigin()
+/** Reason for creating a new channel or a splice. */
+// @formatter:off
+sealed class Origin {
+    abstract val amount: MilliSatoshi
+    abstract val serviceFee: MilliSatoshi
+    abstract val miningFee: Satoshi
+    data class PayToOpenOrigin(val paymentHash: ByteVector32, val fee: Satoshi, override val amount: MilliSatoshi) : Origin() {
+        override val serviceFee: MilliSatoshi = if (fee > 1_000.sat) 1000.sat.toMilliSatoshi() else 0.msat
+        override val miningFee: Satoshi = if (fee > 1_000.sat) fee - 1000.sat else fee
```

Comment:
Comment from @t-bast on parent PR: https://github.com/ACINQ/lightning-kmp/pull/430#discussion_r1154588805

> Can you detail what you're doing here? We're not going to hard-code these values, are we?

Reply:
Cleaned up in a later iteration, see [1affc9e](https://github.com/ACINQ/lightning-kmp/pull/436/commits/1affc9eaa45d4050caf5363c08ef0c7f457a51b3).